### PR TITLE
Rename table experience_skills to experience_skill

### LINKED ISSUE #2403: Rename experience_skills table
The pivot table `experience_skills` should have both model names singular to comply with current convention.  Postgres has table rename functionality so this should be pretty simple.

### PR DESCRIPTION
**Pull request by mnigh**
Resolves #2403.

### Steps to Test
1. In a browser, view [experience_skills table structure with Adminer](http://localhost:8080/adminer?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=experience_skills) and make note of the **Indexes** and **Foreign keys**
2. To run the new migration on the database from inside the [docker container](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/README.md#running-with-docker-containers), run `docker-compose exec -w /var/www/html/api php sh -c "php artisan migrate"`.
3. In a browser, view [experience_skill table structure with Adminer](http://localhost:8080/adminer?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=experience_skill) and make note of the **Indexes** and **Foreign keys** (they should be the same as before the migration but any instances of _experience_skills_ will be _experience_skill_)
4. To reverse the migration on the database from inside the [docker container](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/api/README.md#running-with-docker-containers), run `docker-compose exec -w /var/www/html/api php sh -c "php artisan migrate:rollback --step=1"`.
5. In a browser, view [experience_skills table structure with Adminer](http://localhost:8080/adminer?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=experience_skills) and make note of the **Indexes** and **Foreign keys** (they should be the same as before the migration and the reversal of the migration with any instances of _experience_skill_ will be _experience_skills_)

### Important Note
For @tristan-orourke or whoever is releasing this to UAT and production: a backup of the database should be made before this migration is run (and then tested and verified) on UAT and production environments.

### CODE REVIEW

**Review by petertgiles — APPROVED** [review]
Shoot, I just approved #3225 which copied the old version of this code and repeated the mistake. :cry: Oh well, we can use this PR as a guide to fix that too.

This looks great to me.  If you want the backup before this is applied you should probably coordinate with Tristan before merging it.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Just to hold off until I'm ready to backup the database.

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**
